pkg/config/config_test.go

```diff
@@ -274,36 +274,59 @@ func TestLoad(t *testing.T) {
 			name:             "global file does not exist and hosts file exist",
 			globalConfigPath: "",
 			hostsConfigPath:  hostsFilePath,
-			wantGitProtocol:  "https",
+			wantGitProtocol:  "",
 			wantToken:        "yyyyyyyyyyyyyyyyyyyy",
 		},
 		{
 			name:             "global file exist and hosts file does not exist",
 			globalConfigPath: globalFilePath,
 			hostsConfigPath:  "",
 			wantGitProtocol:  "ssh",
-			wantGetErr:       true,
+			wantToken:        "",
+		},
+		{
+			name:             "global file does not exist and hosts file does not exist with no fallback",
+			globalConfigPath: "",
+			hostsConfigPath:  "",
+			wantGitProtocol:  "",
+			wantToken:        "",
+		},
+		{
+			name:             "global file does not exist and hosts file does not exist with fallback",
+			globalConfigPath: "",
+			hostsConfigPath:  "",
+			fallback:         ReadFromString(testFullConfig()),
+			wantGitProtocol:  "ssh",
+			wantToken:        "yyyyyyyyyyyyyyyyyyyy",
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			cfg, err := load(tt.globalConfigPath, tt.hostsConfigPath)
+			cfg, err := load(tt.globalConfigPath, tt.hostsConfigPath, tt.fallback)
 			if tt.wantErr {
 				assert.EqualError(t, err, tt.wantErrMsg)
 				return
 			}
 			assert.NoError(t, err)
+
+			fmt.Println(cfg)
+
 			protocol, err := cfg.Get([]string{"git_protocol"})
-			assert.NoError(t, err)
-			assert.Equal(t, tt.wantGitProtocol, protocol)
+			if tt.wantGitProtocol == "" {
+				assert.EqualError(t, err, `could not find key "git_protocol"`)
```

Comment:
I kind of like:

```
var keyNotFoundError *KeyNotFoundError
assert.ErrorsAs(t, err, &keyNotFoundError)
```

String checks on errors are kinda last resort in my mind. That's why i introduced this helper:

```
func requireNoKey(t *testing.T, cfg *Config, keys []string) {
	t.Helper()

	_, err := cfg.Get(keys)
	var keyNotFoundError *KeyNotFoundError
	require.ErrorAs(t, err, &keyNotFoundError)
}
```

Also:

```
func requireKeyWithValue(t *testing.T, cfg *Config, keys []string, value string) {
	t.Helper()

	actual, err := cfg.Get(keys)
	require.NoError(t, err)

	require.Equal(t, value, actual)
}
```

Was handy in other cases.